medusa/search/proper.py

```diff
@@ -334,37 +336,52 @@ def _download_propers(self, proper_list):
 
             main_db_con = db.DBConnection()
             history_results = main_db_con.select(
-                b'SELECT resource FROM history '
+                b'SELECT resource, proper_tags FROM history '
                 b'WHERE showid = ? '
                 b'AND season = ? '
-                b'AND episode = ? '
+                b'AND episode IN ({episodes}) '
                 b'AND quality = ? '
                 b'AND date >= ? '
-                b'AND action in (?, ?, ?, ?)',
-                [cur_proper.indexerid, cur_proper.actual_season, cur_proper.actual_episode, cur_proper.quality,
+                b'AND action IN (?, ?, ?, ?) '
+                b'LIMIT {amount}'.format(
```

Comment:
`LIMIT` because we want Downloaded + the Snatched prior to that. There is no need to get the whole history of an episode.

Reply:
Are you sure that's not needed?
What if you snatched / downloaded [PP'ed] the same episode multiple times in a row?
And it stops at '30 days ago' anyway.

Reply:
It was an attempt to make this chunk of slow code somewhat faster. Checking the history is unreliable anyway, as history data is volatile. I will remove the `LIMIT`.